web/src/invite.ts

```diff
@@ -385,8 +347,22 @@ function open_invite_user_modal(e: JQuery.ClickEvent<Document, undefined>): void
 
         const user_has_email_set = !settings_data.user_email_not_configured();
 
-        set_custom_time_inputs_visibility();
-        set_expires_on_text();
+        settings_ui.set_custom_time_inputs_visibility(
+            $expires_in,
+            custom_expiration_time_unit,
+            custom_expiration_time_input,
+        );
+        let expiration_time_in_minutes;
+        if ($expires_in.val() !== "custom") {
+            expiration_time_in_minutes = Number.parseFloat($expires_in.val()!);
+        } else {
+            expiration_time_in_minutes = util.get_custom_time_in_minutes(
+                custom_expiration_time_unit,
+                custom_expiration_time_input,
+            );
+        }
```

Comment:
This logic doesn't seem to be identical to what we had before, with the new `parseFloat`, for example; can you explain what you're doing here?

Reply:
Earlier `set_expires_on_text` was using `valid_to` function internally only to generate the **expires on text**.
To make things more reusable, I have moved out that part of text generation and passed it as an argument in my modified `set_expires_on_text` (see how this function is defined in this PR).
By doing this we would be able to pass different `"valid till texts"` from different files (like "Expires on {date} at {time}" for invitation modal and "Data to be deleted on {date} at {time}" for upcoming scheduled deletion of realm data feature).
So coming back to `parseFloat`, since `valid_to` requires a number as an argument we need to convert `$expires_in.val()!` (which is a string) to a number and then pass it to `valid_to` to generate `"valid till texts"`

Reply:
Found a related [comment](https://github.com/zulip/zulip/pull/26109/files#r1276139505) from #26109.